Assets/Mirror/Tests/Editor/NetworkBehaviourSerializeTest.cs

```diff
@@ -0,0 +1,308 @@
+using NUnit.Framework;
+using UnityEngine;
+
+// Note: Weaver doesn't run on nested class so so use namespace to group classes instead
+namespace Mirror.Tests.NetworkBehaviourSerialize
+{
+    #region No OnSerialize/OnDeserialize override
+    abstract class AbstractBehaviour : NetworkBehaviour
+    {
+        public readonly SyncListBool syncListInAbstract = new SyncListBool();
+
+        [SyncVar]
+        public int SyncFieldInAbstract;
+    }
+
+    class BehaviourWithSyncVar : NetworkBehaviour
+    {
+        public readonly SyncListBool syncList = new SyncListBool();
+
+        [SyncVar]
+        public int SyncField;
+    }
+
+    class OverrideBehaviourFromSyncVar : AbstractBehaviour
+    {
+
+    }
+
+    class OverrideBehaviourWithSyncVarFromSyncVar : AbstractBehaviour
+    {
+        public readonly SyncListBool syncListInOverride = new SyncListBool();
+
+        [SyncVar]
+        public int SyncFieldInOverride;
+    }
+
+
+    class MiddleClass : AbstractBehaviour
+    {
+        // class with no sync var
```

Comment:
@James-Frowen could you add one more test where MiddleClass and SubClass both have a SyncVar please?

Reply:
Done